src/saturn_engine/utils/error_handling.py

```diff
@@ -0,0 +1,144 @@
+from typing import Optional
+from typing import Type
+from typing import cast
+
+import os
+import re
+from dataclasses import dataclass
+from functools import lru_cache
+from textwrap import dedent
+from types import TracebackType
+
+from saturn_engine.utils import inspect as extra_inspect
+from saturn_engine.worker.executors.bootstrap import RemoteException
+
+
+@dataclass
+class ExceptionDetails:
+    message: str
+    exception_type: Type[BaseException]
+    module: str
+    lineno: int
+
+
+@dataclass
+class ExceptionFilter:
+    message_pattern: Optional[re.Pattern]
+    exception_type: Type[BaseException]
+    module_pattern: Optional[re.Pattern]
+    lineno: int
+
+
+def does_error_match(
+    *, exception: ExceptionDetails, exception_filter: ExceptionFilter
+) -> bool:
+    return (
+        (
+            exception_filter.message_pattern is None
+            or exception_filter.message_pattern.match(exception.message) is not None
+        )
+        and issubclass(exception.exception_type, exception_filter.exception_type)
+        and (
+            exception_filter.module_pattern is None
+            or exception_filter.module_pattern.match(exception.module) is not None
+        )
+        and (
+            exception_filter.lineno == 0 or exception.lineno == exception_filter.lineno
+        )
+    )
+
+
+def resolve_exception(exception: str) -> Type[BaseException]:
+    if "." not in exception:
+        return extra_inspect.import_name(f"builtins.{exception}")
+    return extra_inspect.import_name(exception).__class__
+
+
+def _file_to_module(path: str) -> str:
+    basefile, _ = os.path.splitext(path)
+    return basefile.replace(os.path.sep, ".")
+
+
+def get_exception_details(
+    exc_value: BaseException, exc_traceback: TracebackType
+) -> ExceptionDetails:
+    exc_type = exc_value.__class__
+    if exc_type is RemoteException:
+        exc: RemoteException = cast(RemoteException, exc_value)
+        import_path = ".".join(
+            [
+                exc.remote_traceback.exc_module,
+                exc.remote_traceback.exc_type,
+            ]
+        )
+        exc_type = resolve_exception(import_path)
+
+    return ExceptionDetails(
+        message=str(exc_value),
+        exception_type=exc_type,
+        module=_file_to_module(exc_traceback.tb_frame.f_code.co_filename),
```

Comment:
As per my comment before, I would suggest to create a `TracebackData` from the exception and add it to the detail, ie:

```
if issubclass(exc_type, RemoteException):
  traceback = exc.remote_traceback
else:
  traceback = TracebackData.from_exc_info(exc_type, exc_value, exc_traceback)
```

And than use this traceback here. In case of remote exception, the traceback is not going to be interesting.

Reply:
You must use the remote-`traceback` to compute `module` and `lineno`. In short, `exc_traceback` is pretty much useless.